firmware/lib/Kinematics/src/Kinematics.h

```diff
@@ -0,0 +1,45 @@
+#ifndef KINEMATICS_H
+#define KINEMATICS_H
+
+#include "Arduino.h"
+#include "Timer.h"
+#include "Encoder.h"
+#include "LimitSwitch.h"
+
+#define VELOCITY_HOMING 200
+
+enum status{pressed, depressed};
+enum direction{positive, negative};
+enum segment{accelerate, hold, decelerate};
+
+typedef struct Axis
+{
+    int dir_pin;
+    int step_pin;
+    uint32_t accel;
+    uint32_t vel_max;
+    uint32_t vel;
+    int32_t vel_profile_cur_trap[3]; // defined in counts accelerate, hold, decelerate
+    int tragectory_segment;
```

Comment:
After typedefing the enums, replace `int` with `Segment`

Reply:
so, this isnt actually an enum, since i didnt think i could ++ enums i just kept it as int, but i compare it to enum for readability